### ISSUE TITLE
Use cran version of R instead the one in ubuntu repo

### ISSUE THREAD

**Issue by danmrc** [tbrown122387/Using-gradeR-for-the-Gradescope-Autograder#6]
Hi,

First of all, I really like the package.

I was trying this out and I got a mysterious bug. It turns out your version won't install the most recent version of R, and this creates all sort of problem. 

To solve this, all that would be needed is to add this before the install step at the steup.sh:

```

apt-get update -qq
apt-get install --no-install-recommends software-properties-common dirmngr
wget -qO- https://cloud.r-project.org/bin/linux/ubuntu/marutter_pubkey.asc | tee -a /etc/apt/trusted.gpg.d/cran_ubuntu_key.asc
add-apt-repository "deb https://cloud.r-project.org/bin/linux/ubuntu $(lsb_release -cs)-cran40/"

```

Which allows us to get the cran version of R instead. This works fine on Ubuntu 22, which is the current default (at least at mine) gradescope. This assumes people are using Ubuntu, but since this is the default and your setup already uses apt, I think it is reasonable

Best

**Comment by tbrown122387** [comment]
Thank you for the suggestion. 

What kinds of bugs are we talking about? 

Replacing the current demo `setup.sh` with this would make it less minimal and would cause longer build times, so there is a tradeoff. I do agree that this is reasonable looking.

If its aim is to fix dependency issues that are specific to specific packages used in specific assignments, then I would file this kind of issue under ordinary adaptations that I leave to individual instructors.

**Comment by danmrc** [comment]
Hi,

The issue I ran into was installing a package. It got a non-zero status because of a requirement of the Matrix package to be of a version above the one that is available on whatever R version is the default for Ubuntu. It is not possible to install the newer version of Matrix in the R version (which was the first thing I tried).

I agree that there is a trade-off here, but this affects Matrix which is a dependency for a ton of packages. Maybe add this to the readme, so first-time users don't need to go and figure out how to solve it?
